numpy/lib/arraypad.py

```diff
@@ -69,8 +69,12 @@ def _linear_ramp(ndim, axis, start, stop, size, reverse=False):
     ramp = ramp.reshape(init_shape)
 
     if size != 0:
+        # In case `stop` < `start` we get a negative gain and need the array
+        # `stop` to have a compatible signed dtype (at least float16)
+        dtype = np.promote_types(stop.dtype, np.float16)
```

Comment:
Perhaps use `ramp.dtype` instead of `np.float16`?

Reply:
That's a good idea. I took it further by finding the most fitting type at the start of the function which is then used everywhere. That might save us a few surprises down the line in case the function will be used elsewhere or "start" and "stop" are swapped.

Off topic: I wish I had known about the existence of np.result_type and similar functions earlier. They seem very useful. :sweat_smile: